### PR TITLE
SHOW DATABASES should work even if current database has been dropped

### LINKED ISSUE #4187: SHOW DATABASES bug
### What version of Materialize are you using?

```
materialized v0.4.3-dev
```

### How did you install Materialize?

* [ ] Docker image
* [ ] Linux release tarball
* [ ] APT package
* [ ] macOS release tarball
* [ ] Homebrew tap
* [X] Built from source

### What was the issue?

`SHOW DATABASES` returns an error if run from a deleted database.

### Is the issue reproducible? If so, please provide reproduction instructions.

Yes!

```
materialize=> create database j;
CREATE DATABASE
materialize=> set database = j;
SET
materialize=> drop database j;
DROP DATABASE
materialize=> show databases;
ERROR:  unknown database 'j'
```

### PR DESCRIPTION
**Pull request by JLDLaughlin**
Fixes #4187

---
This change is [Reviewable](https://reviewable.io/reviews/materializeinc/materialize/4188)

### CODE REVIEW

**Comment by JLDLaughlin** [comment]
Unfortunately, this is a bit more than just updating `handle_show_databases` to use the qualified `mz_catalog.mz_databases` name because a call to `resolve_schema` was returning an `UnknownDatabase` error.

File: test/testdrive/catalog.td [hunk 1]
```diff
@@ -116,18 +128,22 @@ Expected end of statement, found: RESTRICT
...
+1           materialize
 
 # The session database should remain set to the dropped database, but future
 # queries that depend on the session database should fail with sensible error
 # messages.
 > SHOW DATABASE
 d
 ! SELECT * FROM v
-unknown database 'd'
+unknown catalog item 'v'
```
Comment (JLDLaughlin) [inline test/testdrive/catalog.td:146]: @benesch - is this change okay?

File: test/testdrive/catalog.td [hunk 2]
```diff
@@ -54,6 +54,18 @@ id          database
...
+
+> SET DATABASE = j
+
+> DROP DATABASE j
+
+> SHOW DATABASES
+database
+----
+materialize
+
```
Comment (benesch) [inline test/testdrive/catalog.td:68]: Isn't this test case exactly the same as the test on L131? Doesn't seem like we need to test it twice!

File: test/testdrive/catalog.td (hunk 1 above)
Comment (benesch) [reply test/testdrive/catalog.td:146]: 100%! In fact this even seems like a better error message.

**Review by benesch — APPROVED** [review]
👍👍👍

**Pull request merged**